pkg/dmaasbackup/schedule.go

```diff
@@ -77,12 +83,159 @@ func (d *dmaasBackup) generateBackupName(dbkp v1alpha1.DMaaSBackup) string {
 // updateScheduleInfo checks for relevant velero schedule/backup and
 // update the dbkp with schedule/backup details
 func (d *dmaasBackup) updateScheduleInfo(dbkp *v1alpha1.DMaaSBackup) error {
-	// TODO
+	// check if dbkp is having config for non-scheduled backup
+	if yes := isSchedule(*dbkp); !yes {
+		bkp, err := d.backupClient.Get(dbkp.Name, metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to update scheduleInfo")
+		}
+		if bkp.Name == "" {
+			// Backup could have deleted due to expiration of TTL
+			// TODO: Should we clear backupname from dmaasbackup? and/or recreate?
+			return nil
+		}
+		if dbkp.Status.VeleroBackupName == nil {
+			backupName := bkp.Name
+			dbkp.Status.VeleroBackupName = &backupName
+		}
+		return nil
+	}
+
+	// for scheduled backup
+	scheduleList, err := d.scheduleClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", dmaasBackupLabel, dbkp.Name),
+	})
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to update scheduleInfo")
+	}
+
+	// map to track active schedule name
+	activeScheduleMap := make(map[string]bool)
+
+	// update missing schedule in status.VeleroSchedules
+	for _, schedule := range scheduleList.Items {
+		// add schedule name to scheduleMap
+		activeScheduleMap[schedule.Name] = true
+
+		schDetail := getVeleroScheduleDetails(schedule.Name, dbkp)
+		if schDetail == nil {
+			pinnedSchedule := schedule
+			appendVeleroSchedule(dbkp, &pinnedSchedule)
+			continue
+		}
+
+		// check if schedule status is active or not
+		// if not then we need to set it to active since schedule resource exist
+		if schDetail.Status != v1alpha1.Active {
+			schDetail.Status = v1alpha1.Active
+		}
+	}
+
+	// update veleroschedule status
+	for index, schedule := range dbkp.Status.VeleroSchedules {
+		switch schedule.Status {
+		case v1alpha1.Active:
+			_, isActive := activeScheduleMap[schedule.ScheduleName]
+			if !isActive {
+				dbkp.Status.VeleroSchedules[index].Status = v1alpha1.Deleted
+			}
+		}
+	}
 	return nil
 }
 
 // cleanupOldSchedule remove the old schedules if fullbackup retention mentioned
 func (d *dmaasBackup) cleanupOldSchedule(dbkp *v1alpha1.DMaaSBackup) error {
```

Comment:
similar to comments on updateScheduleInfo, for handling a case which happens rarely, lot of code is getting executed regularly.